src/main/java/com/pusher/client/example/PrivateEncryptedChannelExampleApp.java

```diff
@@ -43,9 +42,16 @@ private PrivateEncryptedChannelExampleApp(final String[] args) {
         channel = pusher.subscribePrivateEncrypted(channelName, this, eventName);
 
         // Keep main thread asleep while we watch for events or application will terminate
-        while (true) {
+        for (int i = 0; ; i++) {
             try {
-                Thread.sleep(1000);
+                Thread.sleep(5000);
+                pusher.disconnect(); // to test clearing of shared secret (via tmp log)
+                pusher.connect(this);
+                Thread.sleep(5000);
+                pusher.unsubscribe(channelName); // to test clearing of shared secret (via tmp log)
+                if (i % 2 == 0) { // to test disconnect on both unsubscribed/subscribed
```

Comment:
I don't think this Example app now models a real life use - it's trying to be a test and is not useful as the first example of Private Encrypted Channels people see. Either we need to refactor this to be clearer exactly what's happening and why we're disconnecting/reconnecting every 5 seconds, or maybe the easier solution is to make a new Private Encrypted Channel Example app that showcases how people handle disconnects - maybe this can be procedurally driven and not require a while loop.

Reply:
thanks! as discussed reverted to not complicate the encrypted channel example, making it more of a test app: 95d5ca30c758f62defb5dc7ca7c687c54cdbfc31